PR: https://github.com/odoo/odoo/pull/209837

From: e53e71f9b80e937690365024f8790fd69b30c7b1
From: sako-odoo
Date: 2025-08-08 16:31:12

Breaking data model changes scores: del:5 + add:0, change matches:
-    is_returned = fields.Boolean(
-    is_repairable = fields.Boolean(
-    return_type_of_ids = fields.One2many('stock.picking.type', 'return_picking_type_id')
-    is_repairable = fields.Boolean(compute='_compute_is_repairable')

Total Changes: 34

[IMP] repair: remove repair option from operation type

* Removed the “Repair Order from Return” checkbox from operation type
  configuration.

* Added a new "Create Repair" server action directly on stock transfers,
  offering the same functionality.

* This change simplifies the process, allowing users to create repair orders
  without needing to configure operation types in advance. Users can initiate
  repair orders directly from transfers via a clear action button, enhancing
  usability and simplifying setup.

Task ID: [4778453](https://www.odoo.com/odoo/project/966/tasks/4778453)

closes odoo/odoo#209837

Related: odoo/enterprise#85448
Related: odoo/upgrade#7681
Signed-off-by: Steve Van Essche <svs@odoo.com>

================================= pseudo patch: =================================

--- a/addons/repair/models/repair.py
+++ b/addons/repair/models/repair.py
@@ -178,12 +178,9 @@ class RepairOrder(models.Model):
 
     # Return Binding
     picking_id = fields.Many2one(
-        'stock.picking', 'Return', check_company=True, index='btree_not_null',
+        'stock.picking', 'Transfer', check_company=True, index='btree_not_null',
         domain="[('return_id', '!=', False), ('product_id', '=?', product_id)]",
-        copy=False, help="Return Order from which the product to be repaired comes from.")
-    is_returned = fields.Boolean(
-        "Returned", compute='_compute_is_returned',
-        help="True if this repair is linked to a Return Order and the order is 'Done'. False otherwise.")
+        copy=False, help="Transfer from which the product to be repaired is picked")
     picking_product_ids = fields.One2many('product.product', compute='_compute_picking_product_ids')
     picking_product_id = fields.Many2one(related="picking_id.product_id")
     allowed_lot_ids = fields.One2many('stock.lot', compute='_compute_allowed_lot_ids')
@@ -320,12 +317,6 @@ class RepairOrder(models.Model):
             elif repair.parts_availability_state == 'late':
                 repair.is_parts_late = True
 
-    @api.depends('picking_id', 'picking_id.state')
-    def _compute_is_returned(self):
-        self.is_returned = False
-        returned = self.filtered(lambda r: r.picking_id and r.picking_id.state == 'done')
-        returned.is_returned = True
-
     @api.depends('move_ids.quantity', 'move_ids.product_uom_qty', 'move_ids.product_uom.rounding')
     def _compute_has_uncomplete_moves(self):
         for repair in self:

--- a/addons/repair/models/stock_picking.py
+++ b/addons/repair/models/stock_picking.py
@@ -41,11 +41,6 @@ class StockPickingType(models.Model):
         check_company=True, store=True, readonly=False, precompute=True,
         help="This is the default recycle destination location when you create a repair order with this operation type.")
 
-    is_repairable = fields.Boolean(
-        'Create Repair Orders from Returns',
-        compute='_compute_is_repairable', store=True, readonly=False, default=False,
-        help="If ticked, you will be able to directly create repair orders from a return.")
-    return_type_of_ids = fields.One2many('stock.picking.type', 'return_picking_type_id')
     repair_properties_definition = fields.PropertiesDefinition('Repair Properties')
 
     def _compute_count_repair(self):
@@ -101,12 +96,6 @@ class StockPickingType(models.Model):
             pt.count_repair_under_repair = counts[pt.id].get('under_repair')
             pt.count_repair_late = late_repairs.get(pt.id, 0)
 
-    @api.depends('return_type_of_ids', 'code')
-    def _compute_is_repairable(self):
-        for picking_type in self:
-            if not picking_type.return_type_of_ids:
-                picking_type.is_repairable = False  # Reset the user choice as it's no more available.
-
     def _compute_default_location_src_id(self):
         remaining_picking_type = self.env['stock.picking.type']
         for picking_type in self:
@@ -188,15 +177,9 @@ class StockPickingType(models.Model):
 class StockPicking(models.Model):
     _inherit = 'stock.picking'
 
-    is_repairable = fields.Boolean(compute='_compute_is_repairable')
     repair_ids = fields.One2many('repair.order', 'picking_id')
     nbr_repairs = fields.Integer('Number of repairs linked to this picking', compute='_compute_nbr_repairs')
 
-    @api.depends('picking_type_id.is_repairable', 'return_id')
-    def _compute_is_repairable(self):
-        for picking in self:
-            picking.is_repairable = picking.picking_type_id.is_repairable and picking.return_id
-
     @api.depends('repair_ids')
     def _compute_nbr_repairs(self):
         for picking in self:
@@ -205,10 +188,10 @@ class StockPicking(models.Model):
     def action_repair_return(self):
         self.ensure_one()
         ctx = clean_context(self.env.context.copy())
+        warehouse = self.picking_type_id.warehouse_id or self.env.user._get_default_warehouse_id()
         ctx.update({
-            'default_product_location_src_id': self.location_dest_id.id,
             'default_repair_picking_id': self.id,
-            'default_picking_type_id': self.picking_type_id.warehouse_id.repair_type_id.id,
+            'default_picking_type_id': warehouse.repair_type_id.id,
             'default_partner_id': self.partner_id and self.partner_id.id or False,
         })
         return {
